packages/ibm-products/src/components/Datagrid/Datagrid/addons/InlineEdit/InlineEditCell/InlineEditCell.js

```diff
@@ -485,7 +506,7 @@ export const InlineEditCell = ({
           config?.validator?.(cellValue),
       })}
     >
-      {!inEditMode && (
+      {!inEditMode && type !== 'checkbox' && (
```

Comment:
Because on UI `checkbox` will always be visible, so no need to do any enter to edit mode here